### PR TITLE
Add app plugin to demo Grafana instance

### PR DESCRIPTION
**Pull request by connorlindsey**
Provisions a locally running instance of the app plugin which is needed to run the plugin + js agent together for local development.

### CODE REVIEW

**Review by bfmatei — CHANGES_REQUESTED** [review]
I would suggest the following approach extremely similar to the dashboards provisioning:

- have a `infa/grafana/plugins` folder
   - plugins can be symlinked in this folder via `ln -s ../grafana-plugins/kwl-plugin infra/grafana/plugins/kwl-plugin` or similar
   - the folder should have an empty `.gitkeep` file be added to git before anything
   - the folder shoud be added to the appropriate ignore files (`.gitignore`, `.prettierignore`, `.eslintignore`) to ensure that anything we add to this folder in the future will not be committed nor checked by linters/code formatters

- have a `infa/grafana/plugins-provisioning` 
   - this folder should have a `plugins.yaml.sample` file that contains the content of `dev.yaml`
   - add `infra/grafana/plugins-provisioning/plugins.yaml` (note: no `.sample` at the end) to `.gitignore` to avoid having it committed by mistake

- have the following variables in the `.env` file:
   - `GRAFANA_PLUGINS_PATH=/var/lib/grafana/plugins`
   - `GRAFANA_PLUGINS_PATH_LOCAL=./infra/grafana/plugins`
   - `GRAFANA_PLUGINS_PROVISIONING_PATH=/etc/grafana/provisioning/plugins`
   - `GRAFANA_PLUGINS_PROVISIONING_PATH_LOCAL=./infra/grafana/plugins-provisioning`

- have `docker-compose.yaml` file updated with:
   ```
         - '${GRAFANA_PLUGINS_PROVISIONING_PATH_LOCAL}:${GRAFANA_PLUGINS_PROVISIONING_PATH}'
         - '${GRAFANA_PLUGINS_PATH_LOCAL}:${GRAFANA_PLUGINS_PATH}'
   ```

- update the readme with a "how to" about using plugins:
   - symlink the plugin folder
   - create `plugins.yaml` file
   - enjoy

This way we are not forcing users to have the plugin in a specific place in their filesystem and they need to opt-in to use it.

**Comment by connorlindsey** [comment]
@bfmatei is it possible with the current setup to load multiple env files? I know this works with a Nextjs. So we could have .env and .env.local where the local one is gitignored. I was thinking people could override the path so that they aren’t forced to put the plugin in a specific location. I’d prefer that as it seems much simpler than the symlinking setup

**Comment by bfmatei** [comment]
@connorlindsey Unfortunately no :(

`docker compose` mimics `docker run` behaviour and the latter is failing with a non-existing `.env` file.

If we are to declare an empty `.env.local` file and add it to `.gitignore`, it won't work as Git will not ignore the changes.

Also, defining the `plugins.yaml` file without providing a path to the plugin will cause Grafana to not start:
```
Failed to start grafana. error: app provisioning error: plugin not installed: "grafana-kowalski-app"
app provisioning error: plugin not installed: "grafana-kowalski-app"
```

:-/

**Comment by connorlindsey** [comment]
Ahh okay, I'll get that updated today then, thanks for the detailed info on getting it working properly

File: infra/grafana/config/grafana.ini [hunk 1]
```diff
@@ -12,4 +12,7 @@ enabled = true
 default_theme = dark
 
 [feature_toggles]
-enable = tempoServiceGraph tempoSearch tempoBackendSearch traceToMetrics traceqlEditor
+enable = tempoServiceGraph tempoSearch tempoBackendSearch traceToMetrics traceqlEditor topNav
```
Comment (eskirk) [inline infra/grafana/config/grafana.ini:15]: All lowercase for `topnav` to get it enabled 

```suggestion
enable = tempoServiceGraph tempoSearch tempoBackendSearch traceToMetrics traceqlEditor topnav
```

**Review by bfmatei — APPROVED** [review]
lgtm!

File: infra/grafana/plugins-provisioning/plugins.yaml.sample [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: 1
+
+apps:
+  - type: grafana-kowalski-app
```
[the remark this answers is not in the document]
Comment (connorlindsey) [reply infra/grafana/plugins-provisioning/plugins.yaml.sample:4]: Good call, I'll move this to the development guide in the plugin repo.

**Pull request merged**